Tribler/Core/Upgrade/db72_to_pony.py

```diff
@@ -172,49 +174,58 @@ def get_old_torrents(self, personal_channel_only=False, batch_size=10000, offset
                                       (" %i " % self.personal_channel_id)
 
         torrents = []
+        batch_not_empty = False # This is a dumb way to indicate that this batch got zero entries from DB
+
         for tracker_url, channel_id, name, infohash, length, creation_date, torrent_id, category, num_seeders, \
             num_leechers, last_tracker_check in \
                 cursor.execute(
                     self.select_full + personal_channel_filter + " group by infohash" +
                     (" LIMIT " + str(batch_size) + " OFFSET " + str(offset))):
+            batch_not_empty = True
             # check if name is valid unicode data
             try:
                 name = text_type(name)
             except UnicodeDecodeError:
                 continue
 
-            try:
-                if len(base64.decodestring(infohash)) != 20:
-                    continue
-                if not torrent_id or int(torrent_id) == 0:
-                    continue
 
-                if not length:
+            try:
+                if (len(base64.decodestring(infohash)) != 20) or \
+                        (not torrent_id or int(torrent_id) == 0) or \
+                        (not length or (int(length) <= 0) or (int(length) > (1 << 45))) or \
+                        not name:
                     continue
 
                 infohash = base64.decodestring(infohash)
 
+                torrent_date = datetime.datetime.utcfromtimestamp(creation_date or 0)
+                torrent_date = torrent_date if 0 <= time2int(torrent_date) <= self.conversion_start_timestamp_int \
+                    else int2time(0)
                 torrent_dict = {
                     "status": NEW,
                     "infohash": infohash,
                     "size": int(length),
-                    "torrent_date": datetime.datetime.utcfromtimestamp(creation_date or 0),
+                    "torrent_date": torrent_date,
                     "title": name or '',
                     "tags": category or '',
                     "tracker_info": tracker_url or '',
                     "xxx": int(category == u'xxx')}
                 if not sign:
                     torrent_dict.update({"origin_id": infohash_to_id(channel_id)})
+                seeders = int(num_seeders or 0)
+                leechers = int(num_leechers or 0)
+                last_tracker_check = int(last_tracker_check or 0)
                 health_dict = {
-                    "seeders": int(num_seeders or 0),
-                    "leechers": int(num_leechers or 0),
-                    "last_check": int(last_tracker_check or 0)}
+                    "seeders": seeders,
+                    "leechers": leechers,
+                    "last_check": last_tracker_check
+                } if (last_tracker_check >= 0 and seeders >= 0 and leechers >= 0) else None
```

Comment:
Putting `and` might not be the correct condition here

Reply:
Why so? The reason for this check is to only convert health entries that have correct (non-negative) attribute values.

Reply:
You're right.